nncf/common/pruning/mask_propagation.py

```diff
@@ -77,6 +77,9 @@ def identity_mask_propagation(node: NNCFNode, graph: NNCFGraph):
     Propagates input mask through nx_node.
     """
     input_masks = get_input_masks(node, graph)
+    if not input_masks:
+        # In case for disconnected NNCFGraph
+        input_masks = [None]
```

Comment:
Before if  `input_masks == []` nncf crashed with assert, assert len(input_masks) == 1. 
This is not affect all supported models, only models with disconnected graph.

Reply:
How did you know? Did you check it manually? Don't think we should do it in that way further: it's time consuming and not reliable. 
IMHO, there should be automatic tests for `ModelPruner` and `MaskPropagationAlgorithm`, checking propagation of masks and covering corner cases like this.

Reply:
About tests, yes, i will add it.

> How did you know? Did you check it manually? Don't think we should do it in that way further: manual testing is time consuming and not reliable.

On current moment models with disconnected graph and empty input masks can not be pruned because nncf crash on `assert len(input_masks) == 1`.  So we have no supported pruned models that affect by this changes.

Reply:
Agreed with @ljaljushkin, all cases (include valid graphs) should be covered by tests.